sway/input/seat.c

```diff
@@ -1197,6 +1197,15 @@ void seat_set_focus(struct sway_seat *seat, struct sway_node *node) {
 		// the workspace needs to be arranged
 		arrange_workspace(new_workspace);
 	}
+
+	#if HAVE_XWAYLAND
+	// Some X11 clients minimize without waiting for the compositor to allow it
+	if (container && container->view && container->view->type == SWAY_VIEW_XWAYLAND
```

Comment:
I'm not a huge fan of xwayland specific code bleeding into this file. The other two cases (which are pre-existing and unrelated to this PR) are for the entire server so those are a little harder to mask away.

However, this is directly related to a view so I'd prefer to see this get moved to `sway/desktop/xwayland.c`. I think it makes sense to just add the following to `set_activated`.
```c
if (activated && surface->minimized) {
  wlr_xwayland_surface_set_minimized(surface, false);
}
```

Otherwise `set_minimized` should be added to `sway_view_impl`, but I'd rather not do that unless the previous solution has some oversight.

Reply:
I didn't really like doing this here either, but I guess I just missed that it could be done in `set_activated`, so thanks for pointing that out.